Firebase/Core/FIRLogger.m

```diff
@@ -213,6 +214,115 @@ BOOL getFIRLoggerDebugMode() {
 }
 #endif
 
+#pragma mark - Number of errors and warnings
+
+NSString *FIRLoggerPathInCachesByAppending(NSString *pathToAppend) {
+  NSArray *directories =
+      NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES);
+  if (directories.count == 0) {
+    return nil;
+  }
+  NSString *cacheDir = directories[0];
+  return [cacheDir stringByAppendingPathComponent:pathToAppend];
+}
+
+/**
+ * Read an integer from the specific file, returning kFIRLoggerCountFileNotFound if the file doesn't
+ * exist.
+ */
+NSInteger FIRReadIntegerFromFile(NSString *filePath) {
+  if (!filePath.length) {
+    return kFIRLoggerCountFileNotFound;
+  }
+
+  NSError *error = nil;
+  NSString *fileContents =
+      [NSString stringWithContentsOfFile:filePath encoding:NSUTF8StringEncoding error:&error];
+  if (error) {
+    return kFIRLoggerCountFileNotFound;
+  }
+
+  return [fileContents integerValue];
+}
+
+/**
+ * Read an integer from the specific file, returning YES if the write was successful.
+ */
+BOOL FIRWriteIntegerToFile(NSInteger value, NSString *filePath) {
+  if (!filePath.length) {
+    return NO;
+  }
+
+  NSString *fileContents = [NSString stringWithFormat:@"%ld", (long)value];
+  NSError *error = nil;
+  BOOL success =
+      [fileContents writeToFile:filePath atomically:YES encoding:NSUTF8StringEncoding error:&error];
+  if (error) {
+    FIRLogDebug(kFIRLoggerCore, @"I-COR000029",
+                @"Attempted to write the a log counter to file but failed: %@", error);
+  }
+
+  return success;
+}
+
+/**
+ * Returns the number of errors logged since last being reset. Note: this synchronously reads a file
+ * on the same queue that logging occurs.
+ */
+NSInteger FIRNumberOfErrorsLogged(void) {
```

Comment:
Is this only for testing? I can't find it's used in other places. How often are we reading this synchronously? I'm afraid of blocking the caller, especially on the main queue.